src/components/Datatable/Datatable.vue

```diff
@@ -0,0 +1,280 @@
+<template>
+  <table>
+    <thead class="table-head">
+      <tr v-if="windowWidth <= 1200" class="mobile-sort-container">
+        <th v-if="windowWidth <= 1200" class="mobile-sort">
+          <slot name="header-sort" :handleMobileSort="sortColumns">
+            <Button @click="sortColumns(columns[0])">Sort: A-Z</Button></slot
+          >
+        </th>
+      </tr>
+
+      <tr v-if="windowWidth >= 1200" class="table-row">
+        <th
+          v-for="(column, columnIndex) in columns"
+          class="table-header"
+          :style="{ textAlign: column.align ? column.align : 'start' }"
+          :key="columnIndex"
+          @click="sortColumns(column)"
+        >
+          <div
+            class="icon-container"
+            :style="{ textAlign: column.align ? column.align : 'start' }"
+          >
+            <slot name="column" :column="column">
+              <slot
+                :name="'column-' + columns[columnIndex].value"
+                :column="column"
+              >
+                {{ columns[columnIndex].text }}
+              </slot>
+
+              <i
+                v-if="currentSortDir == 'asc'"
+                class="pi pi-arrow-up-line pi-1x"
+              />
+              <i
+                v-else-if="currentSortDir == 'desc'"
+                class="pi pi-arrow-down-line pi-1x"
+              />
+
+              <div v-else class="rotate-arrow">
+                <i class="pi pi-code-line pi-1x" />
+              </div>
+            </slot>
+          </div>
+        </th>
+      </tr>
+      <tr class="table-row">
+        <td :colspan="Object.keys(columns).length" class="search-input">
```

Comment:
`columns` is an array so we can check the length directly right? This is in the search empty state as well. 

```suggestion
        <td :colspan="columns.length" class="search-input">
```